src/SortProgram.java

```diff
@@ -1,56 +1,77 @@
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Scanner;
 
 public class SortProgram {
 
-    private Scanner myScanner = new Scanner(System.in);
+    private ArrayList<String> myList = new ArrayList<>();
 
     public void startProgram() {
-        ArrayList myList = new ArrayList();
-        addToList(myList);
-        System.out.println(myList+"\n");
-        do {
-            printBitch("[ADD]\n[REMOVE]\n");
-            String menuInput = myScanner.next();
-            if (menuInput.equals("ADD")) {
-                printBitch("What would you like to add, bitch?");
-                String item = myScanner.next();
-                myList.add(item);
-                printBitch(myList);
-            } else if (menuInput.equals("REMOVE")) {
-                printBitch("What would you like to remove, bitch?");
-                String item = myScanner.next();
-                int idx = 0;
-                while (idx < myList.size()) {
-                    if (myList.get(idx) == item) {
-                        myList.remove(idx);
-                    } else {
-                        ++idx;
-                    }
-                }
-                printBitch(myList);
-            } else {
+        printBitch("\nHello, bitch, gimme something to list!\n");
+        addListItem();
+        displayMenu();
+    }
+
+    private void displayMenu() {
+        System.out.println("\nYour list: " + myList);
+        printBitch("\n[A]DD\n[R]EMOVE\n[S]ORT\n[E]XIT");
+        menuSelection(getUserInput());
+    }
+
+    private void menuSelection(String input) {
+        switch (firstLetter(input)) {
+            case "a":
+                addListItem();
+                displayMenu();
+                break;
+            case "r":
+                removeListItem();
+                displayMenu();
+                break;
+            case "s":
+                Collections.sort(myList);
+                displayMenu();
+                break;
+            case "e":
+                endProgram();
+                break;
+            default:
+                printBitch("Invalid selection, try again, bitch.");
+                displayMenu();
                 break;
-            }
-        }while(true);
```

Comment:
Avoid `do while` loops if possible. They can be very useful, but they come with the possibility of creating an infinite loop if you mess up somewhere. Finding the cause of the infinite loop becomes a matter of looking through the scope of the `do while`, instead of simply looking at the loop's own escape condition. This is especially true with `while(true)`.